api/certs/cert.go

```diff
@@ -202,11 +226,36 @@ func prepareCerts(wssCertExists, httpsCertExists bool, wssCertReady, httpsCertRe
 	}
 }
 
-func ChangeToDefault(https, wss bool) {
-	defaultDomain, err := util.GetDefaultDomainFromIP(config.Parameters.CertDomainName, config.Parameters.DefaultTlsDomainTmpl)
-	if err != nil {
-		log.Error(err)
+func SetDefaultDomain(https, wss bool) (bool, error) {
+	var defaultDomain string
+	var err error
+	var defaultCertAvailable bool
+	for _, d := range config.Parameters.DefaultTlsDomainTmpl {
+		defaultDomain, err = util.GetDefaultDomainFromIP(config.Parameters.CertDomainName, d)
+		if err != nil {
+			log.Error(err)
+			continue
+		}
+		ok, err := domainDNSCheck(defaultDomain)
+		if err != nil {
+			log.Error(err)
+			defaultDomain = ""
+			continue
+		}
+		if ok {
+			_, _, certFile, keyFile := GetACMEFileNames(defaultDomain, config.Parameters.CertDirectory)
+			_, err = VerifyLocalCertificate(certFile, keyFile, defaultDomain)
+			if err != nil {
+				log.Error(err)
+				continue
+			}
+		}
+		break
```

Comment:
The logic here is still not right. An example:
domain A B C all have valid dns pointing to the right IP;
if none of A/B/C has valid cert, then we should use A and apply for cert;
if A/C have no valid cert, B has valid cert, then we should use B;